node/test/PublicAPITest.ts

```diff
@@ -7,14 +7,76 @@ import { assert } from 'chai';
 import * as SignalClient from '../index';
 
 describe('SignalClient', () => {
-  it('can generate and serialize PrivateKeys', () => {
-    const a = new SignalClient.PrivateKey();
-    const b = new SignalClient.PrivateKey();
-    assert.equal(a.serialize().length, 32, 'correct length');
-    assert(a.serialize().equals(a.serialize()), 'repeatable');
+  it('ECC signatures work', () => {
+    const priv_a = SignalClient.PrivateKey.generate();
+    const priv_b = SignalClient.PrivateKey.generate();
+    assert.equal(priv_a.serialize().length, 32, 'correct length');
+    assert(priv_a.serialize().equals(priv_a.serialize()), 'repeatable');
     assert(
-      !a.serialize().equals(b.serialize()),
+      !priv_a.serialize().equals(priv_b.serialize()),
       'different for different keys'
     );
+
+    const pub_a = priv_a.public_key();
+    const pub_b = priv_b.public_key();
+
+    const msg = Buffer.from([1, 2, 3]);
+
+    const sig_a = priv_a.sign(msg);
+    assert.equal(sig_a.length, 64, 'expected signature size');
+
+    assert(pub_a.verify(msg, sig_a));
+    assert(!pub_b.verify(msg, sig_a));
+
+    const sig_b = priv_b.sign(msg);
+    assert.equal(sig_b.length, 64, 'expected signature size');
+
+    assert(pub_b.verify(msg, sig_b));
+    assert(!pub_a.verify(msg, sig_b));
+  });
+
+  it('ECC key agreement work', () => {
+    const priv_a = SignalClient.PrivateKey.generate();
+    const priv_b = SignalClient.PrivateKey.generate();
+
+    const pub_a = priv_a.public_key();
+    const pub_b = priv_b.public_key();
+
+    const shared_a = priv_a.agree(pub_b);
+    const shared_b = priv_b.agree(pub_a);
+
+    assert(shared_a.equals(shared_b), 'key agreement works');
+  });
+
+  it('ECC keys roundtrip through serialization', () => {
+    const key = Buffer.alloc(32, 0xab);
+    const priv = SignalClient.PrivateKey.deserialize(key);
+    assert(key.equals(priv.serialize()));
+
+    const pub = priv.public_key();
+    const pub_bytes = pub.serialize();
+    assert.equal(pub_bytes.length, 32 + 1);
+
+    const pub2 = SignalClient.PublicKey.deserialize(pub_bytes);
+
+    assert(pub.serialize().equals(pub2.serialize()));
+  });
+
+  it('decoding invalid ECC key throws an error', () => {
+    const invalid_key = Buffer.alloc(33, 0xab);
+
+    try {
+      SignalClient.PrivateKey.deserialize(invalid_key);
+      assert(false, 'deserialized invalid key');
+    } catch (err) {
+      assert.equal(err.message, 'bad key length <33> for key with type <Djb>');
+    }
```

Comment:
I think this is clearer, shorter, and more idiomatic:

```suggestion
    assert.throws(() => {
      SignalClient.PrivateKey.deserialize(invalid_key);
    }, 'bad key length <33> for key with type <Djb>');
```

Also applies to the assertions below.

Reply:
Changed thanks